controllers/adminController.ts

```diff
@@ -0,0 +1,193 @@
+import { UserSkillDb } from '../models/UserSkillsSchema';
+import { ICsvSkillsErrors, ICsvValidation, IUserSkills } from '../types';
+import { validateHeader, validateRow, validateRowEmail, validateRowUrls } from '../utils/validateUserSkills';
+import { handleEmail } from '../utils/handleEmail';
+import { generateToken } from '../utils/generateToken';
+import { UserDb } from '../models/UserSchema';
+import { registerUserEmailTemplate } from '../templates/registerUserEmailTemplate';
+import { registerHrEmailTemplate } from '../templates/registerHrEmailTemplate';
+import { filterAdmin, filterHr } from '../utils/filterResponse';
+import { randomPassword } from '../utils/randomPassword';
+import { ValidationError } from '../utils/handleError';
+
+export const validateUserSkills = async (req: Request, res: Response, next: NextFunction) => {
+  const csvFile = req.file;
+  if (!csvFile) throw new ValidationError('Nie przesłano żadnego pliku CSV.');
+
+  try {
+    const csv = await readFile(path.join(storageDir(), 'csv', csvFile.filename), 'utf8');
+    const {
+      meta: { fields },
+      data,
+    } = parse(csv, {
+      header: true,
+      skipEmptyLines: true,
+    });
+
+    const headers = validateHeader(fields).filter((header) => header !== null);
+
+    if (headers.length > 0) {
+      return res.status(200).send({ errors: headers });
+    }
+
+    const emailsFromCsv = data.map((data: IUserSkills) => data.email);
+
+    const duplicateEmails: ICsvValidation[] = emailsFromCsv
+      .map((email: string, i: number) => {
+        const duplicate = emailsFromCsv.indexOf(email) !== i;
+        return duplicate ? { row: i + 1, field: email, message: 'Podany email istnieje już w bazie danych.' } : null;
+      })
+      .filter((data) => data !== null);
+
+    const rowsFromCsv = await Promise.all(
+      data.map(async (dataCsv: any, i) => {
+        const errors = {} as ICsvSkillsErrors;
+        const row = i + 1;
+
+        errors.email = await validateRowEmail(dataCsv.email, i + 1, duplicateEmails);
+        errors.courseCompletion = validateRow(dataCsv.courseCompletion, row);
+        errors.courseEngagement = validateRow(dataCsv.courseEngagement, row);
+        errors.teamProjectDegree = validateRow(dataCsv.teamProjectDegree, row);
+        errors.projectDegree = validateRow(dataCsv.projectDegree, row);
+        errors.bonusProjectUrls = validateRowUrls(dataCsv.bonusProjectUrls, row);
+
+        return Object.values(errors).some((error) => error) ? errors : null;
+      })
+    );
+
+    const rows = rowsFromCsv.filter((row) => row !== null);
+
+    if (rows.length > 0) {
+      return res.status(200).send({ errors: rows });
+    }
+
+    res.status(200).send({ ok: true, fileName: csvFile.filename });
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      throw new ValidationError('Nie znaleziono pliku.');
+    }
+    if (csvFile) {
+      await unlink(path.join(storageDir(), 'csv', csvFile.filename));
```

Comment:
Dlaczego to jest tutaj? Nie powinno być w finally? Lepiej, żeby zawsze ten plik na końcu był usuwany.